--- origsrc/flac-1.3.1/configure.ac	2014-12-02 03:09:34.901199000 -0600
+++ src/flac-1.3.1/configure.ac	2014-12-02 03:09:43.703316700 -0600
@@ -164,7 +164,7 @@ AM_CONDITIONAL(FLaC__CPU_PPC_SPE, test "
 
 os_is_windows=no
 case "$host" in
-	*-*-cygwin|*mingw*)
+	*mingw*)
 		CPPFLAGS="-D__MSVCRT_VERSION__=0x0601 $CPPFLAGS"
 		os_is_windows=yes
 		;;
--- origsrc/flac-1.3.1/src/libFLAC/stream_decoder.c	2014-11-26 20:40:37.344426000 -0600
+++ src/flac-1.3.1/src/libFLAC/stream_decoder.c	2014-12-02 03:09:43.709817500 -0600
@@ -1268,7 +1268,7 @@ FILE *get_binary_stdin_(void)
 	_setmode(_fileno(stdin), _O_BINARY);
 #elif defined __CYGWIN__
 	/* almost certainly not needed for any modern Cygwin, but let's be safe... */
-	setmode(_fileno(stdin), _O_BINARY);
+	_setmode(fileno(stdin), _O_BINARY);
 #elif defined __EMX__
 	setmode(fileno(stdin), O_BINARY);
 #endif
--- origsrc/flac-1.3.1/src/libFLAC/stream_encoder.c	2014-11-26 19:19:47.048799900 -0600
+++ src/flac-1.3.1/src/libFLAC/stream_encoder.c	2014-12-02 03:09:43.716318300 -0600
@@ -4516,7 +4516,7 @@ FILE *get_binary_stdout_(void)
 	_setmode(_fileno(stdout), _O_BINARY);
 #elif defined __CYGWIN__
 	/* almost certainly not needed for any modern Cygwin, but let's be safe... */
-	setmode(_fileno(stdout), _O_BINARY);
+	_setmode(fileno(stdout), _O_BINARY);
 #elif defined __EMX__
 	setmode(fileno(stdout), O_BINARY);
 #endif
--- origsrc/flac-1.3.1/src/share/grabbag/file.c	2014-11-26 19:19:47.048799900 -0600
+++ src/flac-1.3.1/src/share/grabbag/file.c	2014-12-02 03:09:43.719318700 -0600
@@ -166,7 +166,7 @@ FILE *grabbag__file_get_binary_stdin(voi
 	_setmode(_fileno(stdin), _O_BINARY);
 #elif defined __CYGWIN__
 	/* almost certainly not needed for any modern Cygwin, but let's be safe... */
-	setmode(_fileno(stdin), _O_BINARY);
+	_setmode(fileno(stdin), _O_BINARY);
 #elif defined __EMX__
 	setmode(fileno(stdin), O_BINARY);
 #endif
@@ -184,7 +184,7 @@ FILE *grabbag__file_get_binary_stdout(vo
 	_setmode(_fileno(stdout), _O_BINARY);
 #elif defined __CYGWIN__
 	/* almost certainly not needed for any modern Cygwin, but let's be safe... */
-	setmode(_fileno(stdout), _O_BINARY);
+	_setmode(fileno(stdout), _O_BINARY);
 #elif defined __EMX__
 	setmode(fileno(stdout), O_BINARY);
 #endif
